scancode

```diff
@@ -105,10 +105,33 @@ _canonicalize_file_path() {
 SCANCODE_BIN="$( realpath "${BASH_SOURCE[0]}" )"
 SCANCODE_ROOT_DIR="$( cd "$( dirname "${SCANCODE_BIN}" )" && pwd )"
 
+SCANCODE_CONFIG_DIR="etc/conf"
+
+GIT_STATUS=$(git -C "$SCANCODE_ROOT_DIR" status --porcelain 2> /dev/null)
+if [ $? -eq 0 ]; then
+    # This is a Git working tree, check whether it is clean.
+    if [ -z "$GIT_STATUS" ]; then
+        # This is a clean working tree, check whether configure has been run
+        # before for this commit.
+        GIT_COMMIT_CONFIGURED=$(cat "$SCANCODE_ROOT_DIR/SCANCODE_GIT_MODE" 2> /dev/null)
+        GIT_COMMIT_HEAD=$(git -C "$SCANCODE_ROOT_DIR" rev-parse HEAD)
+        if [ "$GIT_COMMIT_CONFIGURED" != "$GIT_COMMIT_HEAD" ]; then
+            echo "* (Re-)Configuring ScanCode because Git commits have changed..."
+            CONFIGURE_QUIET=1 $SCANCODE_ROOT_DIR/configure $SCANCODE_CONFIG_DIR
```

Comment:
IMHO to be really safe, you should first issue a `configure --clean` but let see first if this is enough.
There could be cases where files have been removed and there could be dangling .pyc files left over in some rare cases

Reply:
FWIW, I tried to avoid a full clean by doing e.g. 773bffa.